packages/uiweb/src/lib/hooks/useSupportChatStream.ts

```diff
@@ -0,0 +1,104 @@
+import { useEffect, useState } from 'react';
+import {
+  createSocketConnection,
+  EVENTS
+} from '@pushprotocol/socket';
+import {  ENV } from '../config';
+import { CONSTANTS, PushAPI, SignerType } from '@pushprotocol/restapi';
+// eslint-disable-next-line @nrwl/nx/enforce-module-boundaries
+import { PushStream } from 'packages/restapi/src/lib/pushstream/PushStream';
+
+
+export type SupportChatStreamHookOption = {
+  account?: string | null,
+  env?: ENV,
+  pushUser: PushAPI ,
+  supportAddress: string | null,
+  signer?: SignerType | null
+};
+
+export const useSupportChatStream =({ account, env , pushUser, supportAddress, signer }: SupportChatStreamHookOption) => {
+  
+  const [messagesSinceLastConnection, setMessagesSinceLastConnection] = useState<any>('');
+  const [isSupportChatStreamConnected, setIsSupportChatStreamConnected] = useState<boolean>(false);
+
+  
+  const attachListenersAndConnect = async (stream: PushStream) => {
+    console.warn('\n--> addChatSocketEvents');
+    stream.on(CONSTANTS.STREAM.CONNECT, (err:Error) => {
+      console.log('CONNECTED: ', err);
+      setIsSupportChatStreamConnected(true);
+    });
+
+
+    stream.on(CONSTANTS.STREAM.DISCONNECT, (err:Error) => {
+      console.log('DIS-CONNECTED: ',err);
+      setIsSupportChatStreamConnected(false);
+    });
+
+
+
+    console.log('\t-->will attach eachMessage event now');
+    stream.on(CONSTANTS.STREAM.CHAT, (message: any) => {
+
+  
+      setMessagesSinceLastConnection(message);
```

Comment:
Also have you tested it out, because the message type is different in new stream